cmd/create/cluster/cmd.go

```diff
@@ -1327,7 +1339,7 @@ func run(cmd *cobra.Command, _ []string) {
 		}
 
 		// Validate subnets in the case the user has provided them using the `args.subnets`
-		if subnetsProvided {
+		if subnetsProvided && !args.hostedClusterEnabled {
```

Comment:
We still need to validate subnet right? whether it is just private?

Reply:
we should have interactive support

Reply:
Also can you add the validation or enforce BYO VPC if it is a hosted cluster

Reply:
The current code in CS, only needs the private subnet id; it then fetches the VPC ID and we randomly chooses a public subnet for the default node pool. Had a chat with @vkareh and we decided this might cause surprises, and we need to modify this behavior. The next patch will have the fix for this, as well as the VPC validation.

Reply:
Why do we need a public subnet for a private link cluster? @vkareh @samira-barouti

Reply:
The OAuth server endpoint is always public (even for the private clusters)
Ref: https://docs.google.com/document/d/1egszbmAMkwI1_ZXgVGuAtCX5wEhGiorq9cMBoS9X08A/edit#heading=h.dxjx8ylhdkac